spatialos-sdk/src/worker/locator.rs

```diff
@@ -264,3 +370,347 @@ pub(crate) extern "C" fn queue_status_callback_handler(
         callback(&Err(str.to_string_lossy().to_string())) as u8
     }
 }
+
+pub struct PlayerIdentityTokenRequest {
+    pub dev_auth_token: CString,
+    pub player_id: CString,
+    pub duration_seconds: Option<u32>,
+    pub display_name: Option<CString>,
+    pub metadata: Option<CString>,
+    pub use_insecure_connection: bool,
+}
+
+impl PlayerIdentityTokenRequest {
+    pub fn new<S: AsRef<str>, T: AsRef<str>>(dev_auth_token: S, player_id: T) -> Self {
+        PlayerIdentityTokenRequest {
+            dev_auth_token: CString::new(dev_auth_token.as_ref())
+                .expect("`dev_auth_token` contained a null byte"),
+            player_id: CString::new(player_id.as_ref()).expect("`player_id` contained a null byte"),
+            duration_seconds: None,
+            display_name: None,
+            metadata: None,
+            use_insecure_connection: false,
+        }
+    }
+
+    pub fn with_duration_secs(mut self, seconds: u32) -> Self {
+        self.duration_seconds = Some(seconds);
+        self
+    }
+
+    pub fn with_display_name<S: AsRef<str>>(mut self, display_name: S) -> Self {
+        self.display_name = Some(
+            CString::new(display_name.as_ref()).expect("`display_name` contained a null byte"),
+        );
+        self
+    }
+
+    pub fn with_metadata<S: AsRef<str>>(mut self, metadata: S) -> Self {
+        self.metadata =
+            Some(CString::new(metadata.as_ref()).expect("`metadata` contained a null bytes"));
+        self
+    }
+
+    pub fn with_insecure_connection(mut self) -> Self {
+        self.use_insecure_connection = true;
+        self
+    }
+
+    fn to_worker_sdk(&self) -> Worker_Alpha_PlayerIdentityTokenRequest {
+        Worker_Alpha_PlayerIdentityTokenRequest {
+            development_authentication_token: self.dev_auth_token.as_ptr(),
+            player_id: self.player_id.as_ptr(),
+            duration_seconds: match self.duration_seconds {
+                Some(ref value) => value as *const _ as *mut u32, // TODO: Remove double cast when C SDK is fixed.
```

Comment:
This TODO might be able to be resolved now 😄

Reply:
I was able to remove the 2nd half of it (I guess the C API forgot to make it const)